datafusion/core/src/execution/context.rs

```diff
@@ -1228,6 +1228,11 @@ pub struct SessionConfig {
     pub collect_statistics: bool,
     /// Should DataFusion optimizer run a top down process to reorder the join keys
     pub top_down_join_key_reordering: bool,
+    /// Should DataFusion optimizer prefer HashJoin over SortMergeJoin.
+    /// HashJoin can work more efficently than SortMergeJoin but consumes more memory.
+    pub prefer_hash_join: bool,
+    /// The maximum estimated size in bytes for the left input a hash join will be collected into one partition
+    pub hash_join_collect_left_threshold: usize,
```

Comment:
I recommend calling this seetting something that doesn't have 'left' and 'right' as that can get confusing.

How about `hash_join_single_partition_threshold`?

Reply:
Sure.